### DISCUSSION TITLE
what is the direction output in traj.to_traj_gdf()?

### DISCUSSION THREAD

**Discussion by raybellwaves** [anitagraser/movingpandas#210] [Q&A] [answered]
I find returning the direction with `traj.to_traj_gdf()` confusing. Is it mean direction over the entire linestring? or direction from start point to end point?

Source code here: https://github.com/anitagraser/movingpandas/blob/master/movingpandas/trajectory.py#L384

```
import geopandas as gpd
import pandas as pd
from shapely.geometry import Point
import movingpandas as mpd

df = pd.DataFrame(
    {
        "t": pd.date_range("2022-01-01", periods=5),
        "geometry": [Point(0, 0), Point(0, 1), Point(1, 1), Point(1, 0), Point(0, 0)],
    }
).set_index("t")
gdf = gpd.GeoDataFrame(df, crs=31256)
mpd.Trajectory(gdf, 1).to_traj_gdf()
```

![Screen Shot 2022-03-15 at 4 17 51 PM](https://user-images.githubusercontent.com/17162724/158464476-330d21b7-b586-4a48-bcae-1018b5f580db.png)

**Comment by anitagraser** [comment, marked as answer]
Yes, Trajectory.get_direction() computes direction based on start and end point:
https://github.com/anitagraser/movingpandas/blob/cea55b79f23cf84e1b2c86a222f4f9e1c4c078ec/movingpandas/trajectory.py#L664-L670
